File: js/examples/chat-chain/chain.mjs
import { ChatOpenAI } from "langchain/chat_models/openai";
import { LLMChain } from "langchain/chains";
import {
  ChatPromptTemplate,
  SystemMessagePromptTemplate,
  HumanMessagePromptTemplate
} from "langchain/prompts";

import dotenv from 'dotenv'
dotenv.config({path: '/workspaces/langchain-tutorial/.env' })

const systemMessagePrompt = SystemMessagePromptTemplate.fromTemplate("Eres un {profesion} poco servicial que responde de manera mordaz a todas las solicitudes");
const humanMessagePrompt = HumanMessagePromptTemplate.fromTemplate("{input}")
const chatPrompt = ChatPromptTemplate.fromPromptMessages([systemMessagePrompt, humanMessagePrompt])

// intenta experimentar con el temperature parameter - como cambia el output?
const chat = new ChatOpenAI({});

// ver si se puede anadir memory al chain - https://js.langchain.com/docs/modules/memory/how_to/buffer

const chain = new LLMChain({
  llm: chat,
  prompt: chatPrompt,
});

const result = await chain.call({
    profesion: "camarero",
    input: "Hola, me gustaría pedir una cerveza"
});

console.log(result)

File: python/examples/reAct-agent/chatChain/chain.py
from langchain.chat_models import ChatOpenAI
from langchain.chains import LLMChain
from dotenv import load_dotenv
load_dotenv()

from langchain import PromptTemplate
from langchain.prompts.chat import (
    ChatPromptTemplate,
    SystemMessagePromptTemplate,
    AIMessagePromptTemplate,
    HumanMessagePromptTemplate,
)

system_message_prompt = SystemMessagePromptTemplate.from_template("""\
Eres un {profesion} poco servicial que responde de manera mordaz a todas las solicitudes..\
""")
                                                                  
human_message_prompt = HumanMessagePromptTemplate.from_template("{input}")

chat_prompt = ChatPromptTemplate.from_messages([system_message_prompt, human_message_prompt])

chat = ChatOpenAI()

chat_chain = LLMChain(llm=chat, prompt=chat_prompt)

# Intenta ver si puedes anadir memory al chat chain.

if __name__ == "__main__":
    #response = chat(chat_prompt.format_prompt(profesion="camarero", input="Hola, no me gusto la comida").to_messages())
    response = chat_chain({'profesion':'camaerero', 'input':'Hola, no me gusta la comida que me ha servido'})
    print(type(response))
    print(response)